docker-compose.yml

```diff
@@ -155,6 +155,43 @@ services:
       - gitbase
       - bblfsh-web
 
+  sourced-ui-celery:
+    image: srcd/sourced-ui:latest
+    restart: unless-stopped
+    environment:
+      SYNC_MODE: ${GITBASE_SIVA}
+      ADMIN_LOGIN: admin
+      ADMIN_FIRST_NAME: admin
+      ADMIN_LAST_NAME: admin
+      ADMIN_EMAIL: admin@example.com
+      ADMIN_PASSWORD: admin
+      POSTGRES_DB: superset
+      POSTGRES_USER: superset
+      POSTGRES_PASSWORD: superset
+      POSTGRES_HOST: postgres
+      POSTGRES_PORT: 5432
+      REDIS_HOST: redis
+      REDIS_PORT: 6379
+      GITBASE_DB: gitbase
+      GITBASE_USER: root
+      GITBASE_PASSWORD:
+      GITBASE_HOST: gitbase
+      GITBASE_PORT: 3306
+      METADATA_DB: metadata
+      METADATA_USER: metadata
+      METADATA_PASSWORD: metadata
+      METADATA_HOST: metadatadb
+      METADATA_PORT: 5432
+      BBLFSH_WEB_HOST: bblfsh-web
+      BBLFSH_WEB_PORT: 8080
+      SUPERSET_ENV: celery
+    depends_on:
+      - postgres
+      - metadatadb
+      - redis
+      - gitbase
+      - sourced-ui
```

Comment:
Shouldn't it be the opposite? I mean `sourced-ui` depending on `sourced-ui-celery`?

Reply:
I'm not sure about it.
My reasoning was: `sourced-ui` container runs bootstrap script and celery wouldn't be able to do anything unless that script has executed.
Another reason: web server can work without celery (some task would just wait and show progress) while celery can't work without the web server (as server creates tasks).

wdyt?

Reply:
I think that by doing the opposite we can avoid that tasks submitted by the webserver need to wait for celery to be ready, and celery, on the other hand, will just idle. The only thing that could be a problem is whether superset has some cron job running on celery.

Reply:
I'm still not sure about it but changed anyway.

Reply:
I've just seen that superset provides some utilities for warming up charts caches that would ru by configuring celerybeat, so celery in that case would need the db to be ready. So I think it's better as you did before.

Reply:
reverted revert 😅

Reply:
Could you help me to understand why the following is not correct?
- the webserver depends on celery
- celery depends on db (from the last @se7entyse7en's message)

And from first @smacker's comment:
What can the webserver do if celery is not yet ready? Is celery that slow to be started, that would be better if other pieces are started before?
Also, about "_celery can't work without the webserver_"; doesn't it happens the same with Redis, for example? What is Redis doing before the webserver starts? it's just idle till the webserver starts using it, but it's not a reason to consider that Redis depends on the webserver, is it?

Reply:
> Could you help me to understand why the following is not correct?

sure! 😄 

> the webserver depends on celery

The webservers for running some queries, for example, those from SQL Lab use celery. So it just sends the task to celery which is now running in a separate container. I don't know how much familiar you are with celery, but this is also why celery needs the same code of the webserver, because the webserver sends the task as serialized which assumes the code is the same on the worker.

> celery depends on db

It's not properly that celery depends on the db, but the tasks sent by the webserver depends on the db being already initialized, and the initialization is done by the webserver.

> What can the webserver do if celery is not yet ready?

Can do everything except sending tasks (so running some queries) to celery.

> Is celery that slow to be started, that would be better if other pieces are started before?

I don't think that it's slow. My only concern was to have the right dependency logic.

> Also, about "celery can't work without the webserver"; doesn't it happens the same with Redis, for example? What is Redis doing before the webserver starts? it's just idle till the webserver starts using it, but it's not a reason to consider that Redis depends on the webserver, is it?

As I said, it's not celery itself to depends on the webserver, but what's running on int.

Given that you mentioned also redis, I'd add that celery also depends on redis. This is because celery is configured to use redis as broker.

Reply:
Great answer Marvin. Thanks. Just a few clarifications:
- Celery and the webserver share absolutely the same code. To simplify pseudo-code: you can call `execute_query(query)` or `execute_query(query).async()`. It is the same function but the first call will be executed where you called it (in the webserver). And the second in celery worker.
- Webserver can send tasks to celery without celery being up. They will just hang in the queue until celery is ready
- Celery isn't slow

Reply:
oh. also nothing really breaks if we do one way or another. it's just about making it logically right

Reply:
> Given that you mentioned also redis, I'd add that celery also depends on redis. This is because celery is configured to use redis as broker.

Can you open issue or PR please?

Reply:
hm. From the diff:
```yaml
  sourced-ui-celery:
    image: srcd/sourced-ui:v0.7.0
    restart: unless-stopped
    environment:
      <<: *superset-env
      SUPERSET_ENV: celery
    depends_on:
      - postgres
      - metadatadb
      - redis
```

Reply:
> I'd add that celery also depends on redis.

yup I didn't mean that it was missing 😄